code/game/mecha/mech_fabricator.dm

```diff
@@ -182,7 +182,13 @@
 	desc = initial(desc)
 
 	var/obj/item/I = new D.build_path
-	I.loc = get_step(src,SOUTH)
+	if(D.locked)
+		var/obj/item/weapon/storage/lockbox/large/L = new /obj/item/weapon/storage/lockbox/large(get_step(src,SOUTH)) //(Don't use capitals in paths, or single letters.
+		I.loc = L
+		L.name += " [initial(D.name)]"

```

Comment:
datum names often differ from the items' actual name---the standard for lockboxes is that they go by the item name of what's placed inside them, and not their datum name; that standard should be kept and consistent.